modules/service/keeper/binding.go

```diff
@@ -123,7 +112,7 @@ func (k Keeper) SetServiceBindingForGenesis(
 	k.SetOwner(ctx, provider, owner)
 	k.SetOwnerProvider(ctx, owner, provider)
 
-	pricing, err := k.ParsePricing(ctx, svcBinding.Pricing)
+	pricing, err := types.ParsePricing(svcBinding.Pricing)
```

Comment:
`types.ParsePricing(svcBinding.Pricing)` is best placed before` k.SetServiceBinding(ctx, svcBinding)`, which can reduce gas